tests/test_ssl.py

```diff
@@ -2997,6 +3095,225 @@ def test_wantWriteError(self) -> None:
+                    logger.debug(
+                        "SSL peer closed or empty data, stopping SSL drain."
+                    )
+                    break
+                total_ssl_read += len(data)
+                logger.debug(
+                    f"Read {len(data)} bytes of SSL application data "
+                    f"(total: {total_ssl_read})."
+                )
+                # Reset counter on successful read
+                consecutive_empty_ssl_reads = 0
+
+            except SSL.WantReadError:
+                consecutive_empty_ssl_reads += 1
+                if consecutive_empty_ssl_reads >= 10:
+                    logger.debug(
+                        "No more SSL application data available after "
+                        f"{consecutive_empty_ssl_reads} retries."
+                    )
+                    break
+                logger.debug(
+                    "No SSL data available, waiting... "
+                    f"(attempt {consecutive_empty_ssl_reads})."
+                )
+                time.sleep(0.01)  # Small delay for non-blocking SSL reads
+
+        logger.debug(
+            f"Finished reading from server. Bytes read: {total_ssl_read}."
+        )
+
+    def _perform_moving_buffer_test(
+        self, client: Connection, buffer_size: int, want_bad_retry: bool
+    ) -> bool:
+        """
+        Attempts a retry write with a moving buffer and checks for
+        'bad write retry' error.
+        Returns True if 'bad write retry' occurs, False otherwise.
+        """
+        logger.debug("Phase 3: Performing moving buffer retry test")
+
+        # Attempt retry with different buffer but same size
+        msg2 = b"Z" * buffer_size
+        logger.debug(f"buffer location for msg2 is {id(msg2):#x}")
+        try:
+            bytes_written = client.send(msg2)
+            assert not want_bad_retry, (
+                "_perform_moving_buffer_test() failed as retry succeeded "
+                f"unexpectedly with {bytes_written} bytes written."
+            )
+            return False  # Retry succeeded
+        except SSL.Error as e:
+            reason = get_ssl_error_reason(e)
+            assert reason == "bad write retry", (
+                f"Retry failed with unexpected SSL error: {e!r}({reason})."
+            )
+            logger.debug(f"Got SSL error: {e!r} ({reason}).")
+            return True  # Bad write retry
+
+    def _shutdown_connections(
+        self,
+        client: Connection,
+        server: Connection,
+        client_socket: socket,
+        server_socket: socket,
+    ) -> None:
+        """Helper to safely shut down SSL connections and close sockets."""
+        logger.debug("--- Cleanup: Shutting down connections ---")
+        try:
+            if client:
+                client.shutdown()
+        except Exception:
+            # An exception is usually thrown so it is caught and ignored.
+            pass
```

Comment:
I agree with Alex's comments: catch-all exceptions will inevitably hide unforeseen problems in the future. Additionally, it's best to scope exception handling to a single instruction as more lines within try/except create more unobvious control flow branches. Plus, something like a `contextlib.suppress()` CM with a clear code comment would relay the intent much better:
```suggestion
        if client:
            with contextlib.suppress(OpenSSL.SSL.Error):
                # When closing connections in the test teardown stage, we don't care
                # about possible TLS-level problems as the test was specifically
                # emulating corner case situations pre-shutdown. We just attempt
                # releasing resources if possible and disregard any possibly related
                # problems that may occur at this point.
                client.shutdown()
```